### DISCUSSION TITLE
Why is my GitHub Pages site not updating?

### DISCUSSION THREAD

**Discussion by GrahamHannington** [community/community#142349] [Other Feature Feedback, Questions, & Ideas] [answered]
### Select Topic Area

Question

### Body

A week or so ago I published this GitHub Pages site:

https://grahamhannington.github.io/bigas

using the root folder of the main branch of this personal repo:

https://github.com/GrahamHannington/bigas

That worked. Subsequent commits to the main branch were almost immediately reflected in the GitHub Pages site.

However, as of yesterday, the GitHub Pages site is no longer updating.

I can see several queued actions (workflow runs):

![image](https://github.com/user-attachments/assets/dd49d1ec-ba0f-49ae-bab8-434bbe0909e1)

Some have been stuck since last week. If these get deleted after they timeout, then: whoa, how long is the timeout? Over a week?!

I've tried using the GitHub API to cancel them (just clicking **Cancel run** didn't work), but the API responds "Cannot cancel a workflow re-run that has not yet queued" (even though the UI shows it as "Queued"!).

I've tried updating `index.html`, just to see if that triggered a site update. Nope.

Yesterday, as a workaround, I unpublished and then re-published the site. That worked. But I don't want to have to do that for every update.

**Comment by gitstua** [comment, marked as answer]
Sorry to hear you are having issues here.

I don't work in support but if you have a stuck Actions workflow then these do get deleted after they timeout but you could choose to raise a support ticket and ask if they can delete the workflow runs for you.

**Reply by GrahamHannington** [reply]
Thanks @gitstua! It took me a while to figure out how to get to a page that allowed me to raise a support ticket. The [GitHub Support website](https://support.github.com/) kept taking me in circles back to GitHub Community. Understandable. I'm on a free plan.

I deliberately haven't yet clicked "Mark as answer" on your comment because I want to wait and see how that ticket progresses. I don't want to offer false hope to users who read my question and your comment in the future.

I'll report back.

**Reply by GrahamHannington** [reply]
## In brief

@gitstua recommended that I raise a support ticket. That was good advice, thank you.

In response to that ticket, GitHub Support:

➕ Cancelled the "stuck" runs for me. (I couldn't cancel them myself, even with a _force-cancel_ API request.)

➕➕ Directed me to change my repo's **Settings > (GitHub) Pages >  Build and deployment > Source** from "Deploy from a branch" to "GitHub Actions". That worked. My site now once again automatically updates after updates to the main branch.

➖ Didn't answer my question about why my site was not updating.

## In detail

I was pleasantly surprised by the GitHub Support experience, especially as a non-paying user.

While the Support team member studiously avoided my question about why my site was not updating, they patiently stepped me through changing my repo's GitHub Pages "Source" setting from "Deploy from a branch" (Classic Pages experience) to "GitHub Actions" (Best for using frameworks and customizing your build process). And then, from the two options presented, I selected "Static HTML" (because I don't need Jekyll builds). I used the GitHub UI to commit the resulting `workflows/static.yml` file (verbatim, unedited) to the main branch.

This worked for me: my site is now updating again. 👍

Why was it necessary to make that change? Why did my site stop updating when it was set to the original "Classic Pages experience"? I have no idea.

Bottom line: changing **Settings > (GitHub) Pages >  Build and deployment > Source** from "Deploy from a branch" to "GitHub Actions" worked for me.

**Reply by gitstua** [reply]
👋🏽  Hi @GrahamHannington - so great to hear this is resolved for you - our community is inclusive of all developers so glad you had a great support experience :)

Very much appreciate you following up with your experience and solution to help others in the community 🥇
